napari/_qt/_tests/test_qt_viewer.py

```diff
@@ -675,3 +679,113 @@ def test_create_non_empty_viewer_model(qtbot):
     del viewer
     qtbot.wait(50)
     gc.collect()
+
+
+def test_async_slice_image_on_current_step_change(make_napari_viewer, qtbot):
+    np.random.seed(0)
+    data = np.random.rand(3, 4, 5)
+    viewer = make_napari_viewer()
+    vispy_image = setup_viewer_for_async_slice_image(viewer, data)
+    assert viewer.dims.current_step != (2, 0, 0)
+
+    viewer.dims.current_step = (2, 0, 0)
+
+    wait_until_vispy_image_data_equal(qtbot, vispy_image, data[2, :, :])
+
+
+def test_async_slice_image_on_order_change(make_napari_viewer, qtbot):
+    np.random.seed(0)
+    data = np.random.rand(3, 4, 5)
+    viewer = make_napari_viewer()
+    vispy_image = setup_viewer_for_async_slice_image(viewer, data)
+    assert viewer.dims.order != (1, 0, 2)
+
+    viewer.dims.order = (1, 0, 2)
+
+    wait_until_vispy_image_data_equal(qtbot, vispy_image, data[:, 2, :])
+
+
+def test_async_slice_image_on_ndisplay_change(make_napari_viewer, qtbot):
+    np.random.seed(0)
+    data = np.random.rand(3, 4, 5)
+    viewer = make_napari_viewer()
+    vispy_image = setup_viewer_for_async_slice_image(viewer, data)
+    assert viewer.dims.ndisplay != 3
+
+    viewer.dims.ndisplay = 3
+
+    wait_until_vispy_image_data_equal(qtbot, vispy_image, data)
+
+
+def test_async_slice_multiscale_image_on_pan(make_napari_viewer, qtbot):
```

Comment:
The multi-scale pan/zoom tests get a little ugly and I have to simulate the actual actions, but this is the best I could do now. Suggestions welcome!